openquake/calculators/event_based_risk.py

```diff
@@ -131,14 +133,14 @@ def event_based_risk(riskinput, riskmodel, param, monitor):
     I = param['insured_losses'] + 1
     L = len(riskmodel.lti)
     aids = getattr(riskinput, 'aids', None)
-    A = len(assetcol) if aids is None else len(aids)
     R = riskinput.hazard_getter.num_rlzs
     param['lrs_dt'] = numpy.dtype([('rlzi', U16), ('ratios', (F32, (L * I,)))])
     agg = numpy.zeros((E, R, L * I), F32)
     result = dict(assratios=[], lrs_idx=AccumDict(accum=[]), aids=aids)
     if param['avg_losses']:
-        # dict (l, r) -> A losses
-        result['avglosses'] = AccumDict(accum=numpy.zeros(A, F64))
+        # dict (l, r) -> loss_by_aid
+        result['avglosses'] = AccumDict(accum=numpy.zeros(len(assetcol), F64)
+                                        if aids is None else {})
```

Comment:
Please add a comment describing the different cases

Reply:
Done